src/main/java/org/codehaus/groovy/classgen/Verifier.java

```diff
@@ -302,6 +310,12 @@ private void checkFinalVariables(final ClassNode node) {
         visitor.visitClass(node);
     }
 
+    private void checkDeadCode(final ClassNode node) {
+        if (null == sourceUnit) return;
+        GroovyClassVisitor visitor = new DeadCodeAnalyzer(sourceUnit);
+        visitor.visitClass(node);
```

Comment:
I think you can get to the `SourceUnit` through the `ClassNode` without having to change `Verifier` all around.